src/statusApi.js

```diff
@@ -0,0 +1,131 @@
+const request = require('request');
+
+// private method symbols
+const _request = Symbol('request');
+
+/**
+ * Commands that can be run on the Status app.
+ */
+class StatusApi {
+  constructor({ip, logger, timeout}) {
+    this.url = `http://${ip}:5561`;
+    this.timeout = timeout;
+    this.logger = logger;
+  }
+
+  /**
+   * Sends a request with a JSON body to the Status app.
+   * 
+   * @param {String} endpoint Specifies the command sent to the Staus app.
+   * @param {Object} body JSON data sent with the request.
+   * @param {String} method HTTP method for the reqeust. Valid options are "POST" and "DELETE".
+   * @param {Function} cb Callback called upon response, called with parameters:
+   *  - {Error} Error that occurred during request.
+   *  - {Object} JSON response from the Status app.
+   * 
+   * @returns {void}
+   */
+  [_request](endpoint, body, method, cb) {
+    request({
+      url: this.url + endpoint,
+      method: method,
+      timeout: this.timeout,
+      json: true,
+      body: body
+    }, (error, _response, respBody) => {
+      this.logger.trace(`REQUEST: ${endpoint} ${JSON.stringify(body)}\nRESPONSE: ${JSON.stringify(respBody)}\nERROR: ${error}`);
+      cb(error, respBody);
+    });
+  }
+
+  /**
+   * Pings the Status app.
+   * Expected response: {"message": "Pong!"}
+   * Equivalent to: POST http://<device_ip>:5561/ping
+   * 
+   * @param {Function} cb Callback called upon response, called with parameters:
+   *  - {Error} Error that occurred during request.
+   *  - {Boolean} True if ping was successful and returned {"message": "Pong!"}.
+   * 
+   * @returns {void}
+   */
+  ping(cb) {
+    this[_request]('/ping', {}, "POST", (err, response) => {
+      if(err) return cb(err);
+      if(!response) return cb('No ping response');
+      cb(err, response.message === 'Pong!');
```

Comment:
WHo sends "Pong!"? That's a bit random lol

Reply:
Yeah this is a bit strange. I'm thinking we should just return true if there is a response, and not check the response contents.